lldb/tools/lldb-server/Plugins/AMDGPU/WaveAMDGPU.h

```diff
@@ -0,0 +1,81 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file contains the declaration of the WaveAMDGPU class, which is used
+/// to represent a wave on the GPU.
+///
+//===----------------------------------------------------------------------===//
+
+#ifndef LLDB_TOOLS_LLDB_SERVER_WAVEAMDGPU_H
+#define LLDB_TOOLS_LLDB_SERVER_WAVEAMDGPU_H
+#include "RegisterContextAMDGPU.h"
+#include "lldb/Host/common/NativeThreadProtocol.h"
+#include "lldb/lldb-enumerations.h"
+#include <amd-dbgapi/amd-dbgapi.h>
+#include <memory>
+#include <signal.h>
+
+namespace lldb_private {
+namespace lldb_server {
+class ProcessAMDGPU;
+
+struct DbgApiWaveInfo {
+  amd_dbgapi_wave_state_t state = AMD_DBGAPI_WAVE_STATE_RUN;
+  amd_dbgapi_wave_stop_reasons_t stop_reason = AMD_DBGAPI_WAVE_STOP_REASON_NONE;
+  amd_dbgapi_workgroup_id_t workgroup_id = AMD_DBGAPI_WORKGROUP_NONE;
+  amd_dbgapi_dispatch_id_t dispatch_id = AMD_DBGAPI_DISPATCH_NONE;
+  amd_dbgapi_queue_id_t queue_id = AMD_DBGAPI_QUEUE_NONE;
+  amd_dbgapi_agent_id_t agent_id = AMD_DBGAPI_AGENT_NONE;
+  amd_dbgapi_process_id_t process_id = AMD_DBGAPI_PROCESS_NONE;
```

Comment:
Is amd_dbgapi_process_id_t the process ID of the native process, or is this really a thread ID for the wave? If so, any reason we aren't using this as the `lldb::tid_t` for the wave? IF not, then is this `process_id` needed in each wave since it would be the actual pid of the native process?

Reply:
The `amd_dbgapi_process_id_t` is an internal identifier for the debug api library. We can't use it as the `tid_t` for the wave because it will be same for all waves in the process.

We don't need to store it here since we already have it in the ProcessAMDGPU. I will delete it from here.